Cleanup configs and refactored launch file

AdarshKaran opened the pull request:
1. Removed unused configuration files
2. Introduced dynamic substitution for UR arm controllers
3. Refactored the MoveIt launch file to use `MoveItConfigsBuilder`

AdarshKaran committed:
Cleanup configs, dynamic param substitutions for UR arms and updates to launch file

AdarshKaran committed:
Added use_fake_hardware argument and necessary logic

AdarshKaran commented:
1. Added `use_fake_hardware argument`. Since `scaled_joint_trajectory_controller` does not work with GZ Sim, `joint_trajectory_controller` will be used instead.
2. Renamed `controllers.yaml` to `moveit_controllers.yaml`. `MoveItConfigsBuilder` expects a controllers file with a prefix by default. This will still be overridden via a dictionary substitution from the launch file.
3. Updated README by adding package link and GIFs for ROX and EMROX.

padhupradheep commented:
For some reason, in rolling it is not working for me. 

```
rviz2-3] [INFO] [1741887798.245745016] [rviz2_moveit.moveit.ros.motion_planning_frame]: group ur_manipulator
[rviz2-3] [INFO] [1741887798.245772669] [rviz2_moveit.moveit.ros.motion_planning_frame]: Constructing new MoveGroup connection for group 'ur_manipulator' in namespace ''
[rviz2-3] [INFO] [1741887798.278191820] [rviz2_moveit.moveit.ros.move_group_interface]: Ready to take commands for planning group ur_manipulator.
[rviz2-3] [INFO] [1741887798.279476157] [rviz2_moveit.moveit.ros.motion_planning_frame]: group ur_manipulator
[rviz2-3] [INFO] [1741887798.279561907] [rviz2_moveit.moveit.ros.motion_planning_frame]: group ur_manipulator
[rviz2-3] [INFO] [1741887798.279805316] [interactive_marker_display_99453378390448]: Sending request for interactive markers
[rviz2-3] [INFO] [1741887798.312592125] [interactive_marker_display_99453378390448]: Service response received for initialization
[move_group-2] [INFO] [1741887801.505597661] [move_group.moveit.moveit.ros.move_group.move_action]: MoveGroupMoveAction: Received request
[move_group-2] [INFO] [1741887801.506226190] [move_group.moveit.moveit.ros.move_group.move_action]: executing..
[rviz2-3] [INFO] [1741887801.506977937] [rviz2_moveit.moveit.ros.move_group_interface]: Plan and Execute request accepted
[move_group-2] [INFO] [1741887801.509106607] [move_group.moveit.moveit.ros.move_group.move_action]: Combined planning and execution request received for MoveGroup action. Forwarding to planning and execution pipeline.
[move_group-2] [INFO] [1741887801.509247825] [move_group.moveit.moveit.ros.plan_execution]: Planning attempt 1 of at most 1
[move_group-2] [INFO] [1741887801.509341897] [move_group.moveit.moveit.ros.move_group.capability]: Using planning pipeline 'ompl'
[move_group-2] [INFO] [1741887801.509573016] [move_group]: Calling PlanningRequestAdapter 'ResolveConstraintFrames'
[move_group-2] [INFO] [1741887801.509611769] [move_group]: Calling PlanningRequestAdapter 'ValidateWorkspaceBounds'
[move_group-2] [INFO] [1741887801.509632006] [move_group]: Calling PlanningRequestAdapter 'CheckStartStateBounds'
[move_group-2] [INFO] [1741887801.509680424] [move_group]: Calling PlanningRequestAdapter 'CheckStartStateCollision'
[move_group-2] [INFO] [1741887801.510767224] [move_group.moveit.moveit.planners.ompl.model_based_planning_context]: Planner configuration 'ur_manipulator' will use planner 'geometric::RRTConnect'. Additional configuration parameters will be set when the planner is constructed.
[move_group-2] [INFO] [1741887801.511046240] [move_group]: Calling Planner 'OMPL'
[move_group-2] [INFO] [1741887801.539463833] [move_group]: Calling PlanningResponseAdapter 'AddTimeOptimalParameterization'
[move_group-2] [INFO] [1741887801.552544552] [move_group]: Calling PlanningResponseAdapter 'ValidateSolution'
[move_group-2] [INFO] [1741887801.555165622] [move_group]: Calling PlanningResponseAdapter 'DisplayMotionPath'
[move_group-2] [WARN] [1741887801.555390038] [move_group.moveit.moveit.ros.planning_pipeline]: The planner plugin did not fill out the 'planner_id' field of the MotionPlanResponse. Setting it to the planner ID name of the MotionPlanRequest assuming that the planner plugin does warn you if it does not use the requested planner.
[move_group-2] [INFO] [1741887801.555549293] [move_group.moveit.moveit.plugins.simple_controller_manager]: Returned 2 controllers in list
[move_group-2] [INFO] [1741887801.555578519] [move_group.moveit.moveit.plugins.simple_controller_manager]: Returned 2 controllers in list
[move_group-2] [INFO] [1741887801.555685005] [move_group.moveit.moveit.ros.trajectory_execution_manager]: Validating trajectory with allowed_start_tolerance 0.01
[move_group-2] [INFO] [1741887801.556923320] [move_group.moveit.moveit.ros.trajectory_execution_manager]: Starting trajectory execution ...
[move_group-2] [INFO] [1741887801.557396317] [move_group.moveit.moveit.plugins.simple_controller_manager]: Returned 2 controllers in list
[move_group-2] [INFO] [1741887801.557437278] [move_group.moveit.moveit.plugins.simple_controller_manager]: Returned 2 controllers in list
[move_group-2] [ERROR] [1741887801.557563568] [moveit.simple_controller_manager.follow_joint_trajectory_controller_handle]: Action client not connected to action server: scaled_joint_trajectory_controller/follow_joint_trajectory

```

I use the launch 

`ros2 launch neo_rox_moveit2 neo_ur_moveit.launch.py prefix:=ur10 arm_type:=ur10 use_gz:=True  use_sim_time:=True`

did you do the updates?

AdarshKaran commented:
Could you please try setting the `use_fake_hardware:=true` argument to change the controllers to `joint_trajectory_controller`
https://github.com/neobotix/neo_rox_moveit2/issues/9#issue-2903221996

the error is from this line 
`[move_group-2] [ERROR] [1741887801.557563568] [moveit.simple_controller_manager.follow_joint_trajectory_controller_handle]: Action client not connected to action server: scaled_joint_trajectory_controller/follow_joint_trajectory`

where it's still trying to connect to the `scaled` version. The above argument should fix it.

AdarshKaran committed:
modified use_fake_hardware arg to use_mock_hardware in neo_rox_moveit2 to maintain consistency, added edge case handling for use_mock_hardware argument

AdarshKaran commented:
- modified `use_fake_hardware` arg to `use_mock_hardware` in `neo_ur_moveit.launch.py` to maintain consistency, added edge case handling for `use_mock_hardware` argument

padhupradheep commented:
is this still relevant?

AdarshKaran commented:
we can close this without merging. I have given a new pr #11 that also addresses these changes
